os/kernel/semaphore/sem_trywait.c

```diff
@@ -132,6 +132,10 @@ int sem_trywait(FAR sem_t *sem)
 
 		saved_state = enter_critical_section();
 
+		if ((sem->flags & FLAGS_SIGSEM) != 0) {
+			DEBUGASSERT(sem->semcount < 2);
+		}
```

Comment:
I don't know this is right. When we use a sem for signal purpose, we should not meet the semcount 2 ever? 
I don't think we can guarantee and it's a wrong state.

Reply:
Ok. I added a new flag to be used for mutex only.